editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/internal/ProjectTemplatesActivator.java

```diff
@@ -104,4 +135,73 @@ public ProjectTemplatesSharedImages(Bundle pluginBundle) {
 			addImage(IMAGE_FUSE_ICON, IMAGE_FUSE_ICON);
 		}
 	}
+
+	class POMChangedReporter implements IResourceChangeListener {
+		/*
+		 * (non-Javadoc)
+		 * @see org.eclipse.core.resources.IResourceChangeListener#resourceChanged(org.eclipse.core.resources.IResourceChangeEvent)
+		 */
+		@Override
+		public void resourceChanged(IResourceChangeEvent event) {
+			if (!listenForEvents) return;
+			if (event.getType() == IResourceChangeEvent.POST_CHANGE) {
+				try {
+					event.getDelta().accept(new DeltaPrinter());
+				} catch (CoreException ex) {
+					pluginLog().logError(ex);
+				}
+			}
+		}
+	}
+	
+	class DeltaPrinter implements IResourceDeltaVisitor {
+		
+		private static final String MARKER_ID = ProjectTemplatesActivator.PLUGIN_ID + ".CamelCatalogVersionProblem";
+		
+		/*
+		 * (non-Javadoc)
+		 * @see org.eclipse.core.resources.IResourceDeltaVisitor#visit(org.eclipse.core.resources.IResourceDelta)
+		 */
+      	public boolean visit(IResourceDelta delta) {
+			IResource res = delta.getResource();
+			if (!res.getName().equalsIgnoreCase("pom.xml")) return true;
+			switch (delta.getKind()) {
+				case IResourceDelta.ADDED:
+				case IResourceDelta.CHANGED:
+					validateConfiguredCamelVersionMatchesLoadedCamelVersion(res);
+					break;
+				default: // nothing
+			}
+			return false; 
+      	}
+      	
+      	private void validateConfiguredCamelVersionMatchesLoadedCamelVersion(IResource res) {
+      		Display.getDefault().asyncExec( () -> {
+		  		String loadedCatalogVersion = CamelCatalogCacheManager.getInstance().getCamelModelForProject(res.getProject()).getVersion();
+				String configuredCamelVersionFromPOM = new CamelMavenUtils().getCamelVersionFromMaven(res.getProject());
+				try {
+					listenForEvents = false;
```

Comment:
if the listenForEvents attribute modification is not synchronized i think that we can be in trouble as it is an attribute which is modified in asyncExec

Reply:
good point